code-challenges.test.js

```diff
@@ -32,13 +62,45 @@ var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saf
 
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67 8, 24]
+var nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
-
+//----- Problem Solving -----
+// Create a describe method that lists the name of the function
+// Add a test method that describes what the function does
+// Create 2 variables to hold the expected values
+// Add an expect method that compares the input to the expected values
+
+//----- Syntax -----
+describe("lowestHighest", () => {
+    const expected1 = [-8, 90]
+    const expected2 = [5, 109]
+    it("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+        expect(lowestHighest(nums1)).toEqual(expected1)
+        expect(lowestHighest(nums2)).toEqual(expected2)
+    })
+})
+// test fails with error ReferenceError: lowestHighest is not defined
 
 // b) Create the function that makes the test pass.
 
+//----- Problem Solving -----
+// Create a function called lowestHighest() that takes an array of numbers as a parameter.
+// Create a variable called newArray to hold an empty array.
+// Loop through array and use Math.min() and Math.max() to find the highest and lowest numbers.
+// I'll use unshift() or .push() methods to add the result of Math.min() and Math.max() to newArray
+// Return an array of 2 values being the lowest and highest numbers only.
+
+//----- Syntax -----
+const lowestHighest = (array) => {
+    // let newArray = [] /* Since push and unshift are mutators, I shouldn't need this empty variable right? */
+    for(let i = 0; i < array.length; i++){
+        array.push(Math.min(...array))
+        array.push(Math.max(...array))
+    }
+    return array
+}
+// Fail - Can't figure this one out. I keep getting RangeError: Maximum call stack size exceeded
```

Comment:
I'm pretty sure you're adding two values to the array each loop so it has + 2 values each i++. Infinite array ...